Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -343,6 +346,22 @@ public final class RegistryClient: Cancellable {
                                         signatureBase64Encoded: $0.signatureBase64Encoded,
                                         signatureFormat: $0.signatureFormat
                                     )
+                                },
+                                signingEntity: $0.signing.flatMap {
+                                    guard let signatureData = Data(base64Encoded: $0.signatureBase64Encoded) else {
+                                        return nil
+                                    }
+                                    guard let signatureFormat = SignatureFormat(rawValue: $0.signatureFormat) else {
+                                        return nil
+                                    }
+                                    let configuration = self.configuration.signing(for: package, registry: registry)
+                                    return try? tsc_await { SignatureValidation.extractSigningEntity(
```

Comment:
I think the method itself is async, so we wont need to use tsc_await here?

Reply:
As far as I can see, nothing in `RegistryClient` is async.

Reply:
what I mean is that this method takes a completion handler. so we dont need to use tsc_await, we can chain the handlers

Reply:
Oh, now I get what you mean. We could do that but I think that'll make this code much more complicated tbh. We can have zero and also more than one resource with signing info, so there'll be a bunch of logic for aggregating the results. IMO seems better to just do this for now and eventually replace with real await instead.